modules/basicfuncs/list-funcs.c

```diff
@@ -92,13 +96,48 @@ list_search_pattern_prepare_glob(ListSearchPattern *self)
   return TRUE;
 }
 
+static gboolean
+list_search_pattern_prepare_pcre(ListSearchPattern *self)
+{
+  const gchar *errptr;
+  gint erroffset;
+  gint rc;
+
+  self->pcre = pcre_compile2(self->keyword->str, PCRE_ANCHORED, &rc, &errptr, &erroffset, NULL);
+  if (!self->pcre)
+    {
+      msg_error("Error while compiling regular expression",
+                evt_tag_str("regular_expression", self->keyword->str),
+                evt_tag_str("error_at", &self->keyword->str[erroffset]),
+                evt_tag_int("error_offset", erroffset),
+                evt_tag_str("error_message", errptr),
+                evt_tag_int("error_code", rc));
+      return FALSE;
+    }
+  self->pcre_extra = pcre_study(self->pcre, PCRE_STUDY_JIT_COMPILE, &errptr);
+  if (errptr)
+    {
+      msg_error("Error while optimizing regular expression",
+                evt_tag_str("regular_expression", self->keyword->str),
+                evt_tag_str("error_message", errptr));
+      pcre_free(self->pcre);
+      if (self->pcre_extra)
+        pcre_free_study(self->pcre_extra);
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
 static gboolean
 list_search_pattern_prepare(ListSearchPattern *self, ListSearchMode mode)
 {
   switch (mode)
     {
     case LSM_GLOB:
       return list_search_pattern_prepare_glob(self);
+    case LSM_PCRE:
+      return list_search_pattern_prepare_pcre(self);
```

Comment:
After glob and pcre is introduced I think `$(list-search ) should be a `TEMPLATE_FUNCTION` instead of a `TEMPLATE_FUNCTION_SIMPLE`.

It can have a state where things that need one-time initialization (`pcre_compile`)  can be initialized.

Reply:
Thanks, really good catch! Done :)